### ISSUE TITLE
WebSite Development For Project

### ISSUE THREAD

**Issue by ZigRazor** [ZigRazor/CXXGraph#213]
Now that the project have a good visibility, we need a new interface with the Public.
So we need a new WebSite for the project that for the moment will be delivered in the github-pages.
We need an attractive interface and a section with the documentation.
The in a second moment we can move the algorithms details in a site page.

**Comment by ZigRazor** [comment]
yes, obviously!
I give you the possibilities to select best design for the site.
good work

**Comment by ZigRazor** [comment]
@iks1 are you working on it?

**Comment by ZigRazor** [comment]
Deassigned due to inactivity

**Comment by iks1** [comment]
hello, I was busy on another work, Since I am free right now , I want to devote some time working on it .  If you are okay can we discuss about it . So here are some questions . Do you want it to initialize the website folder in this repository itself or you want to initialize an another repository ? And  I was thinking of using React for this project let me know if you are okay with it , we can deploy it on vercel easily

**Comment by iks1** [comment]
if you are okay i can start toady itself

**Comment by ZigRazor** [comment]
Yes, it's ok.
I think on this repo, if possible with github pages also react is ok.
The branch is gh-page

**Comment by iks1** [comment]
okay then working on it , please reassign

**Comment by iks1** [comment]
if you want i want to share to some samples of documentation websites , let me know which one you want

**Comment by ZigRazor** [comment]
Yes, I appreciate the show of samples.
So we can choose the best for this kind of project.
Thank you in advance

**Comment by iks1** [comment]
okay on it

**Comment by ZigRazor** [comment]
Yes, the branch is deployed, the site can be visited from github pages

**Comment by iks1** [comment]
when I try to run the site locally I am getting the error @ZigRazor  could you please help !
![image](https://github.com/ZigRazor/CXXGraph/assets/94672267/664b9607-6d33-4358-b423-2fff1eb74c1d)

**Comment by iks1** [comment]
@ZigRazor according to me there should be some key points like 
1. What it is and how to install 
2. Different functions and explaining how they work what they are and how to use the different algorithms using some sample codes like this one which is for dijkstras
![image](https://github.com/ZigRazor/CXXGraph/assets/94672267/c7d50e95-6869-4428-8e44-f0e141e2d2a7)

###
And for the documentation I think the layout should be something like this , consisting of three columns, middle one the main content, left side bar the name of topics and right one the related topics , just like the microsoft STL documentation 
![image](https://github.com/ZigRazor/CXXGraph/assets/94672267/23f144bd-1e85-4f7d-83b2-ed48023c4f60)

###
And the left side bar should have subtopics similar to this 
![image](https://github.com/ZigRazor/CXXGraph/assets/94672267/c2b747bf-428f-454d-86fa-045f46549308)
Please let me know your views on it .

**Comment by ZigRazor** [comment]
> when I try to run the site locally I am getting the error @ZigRazor could you please help ! ![image](https://user-images.githubusercontent.com/94672267/253743199-664b9607-6d33-4358-b423-2fff1eb74c1d.png)

I don't know why? look at the deployed site: https://zigrazor.github.io/CXXGraph/

**Comment by ZigRazor** [comment]
> @ZigRazor according to me there should be some key points like
> 
> 1. What it is and how to install
> 2. Different functions and explaining how they work what they are and how to use the different algorithms using some sample codes like this one which is for dijkstras
>    ![image](https://user-images.githubusercontent.com/94672267/253757475-c7d50e95-6869-4428-8e44-f0e141e2d2a7.png)
> 
> And for the documentation I think the layout should be something like this , consisting of three columns, middle one the main content, left side bar the name of topics and right one the related topics , just like the microsoft STL documentation ![image](https://user-images.githubusercontent.com/94672267/253758404-23f144bd-1e85-4f7d-83b2-ed48023c4f60.png)
> 
> And the left side bar should have subtopics similar to this ![image](https://user-images.githubusercontent.com/94672267/253758711-c2b747bf-428f-454d-86fa-045f46549308.png) Please let me know your views on it .

All correct! the idea is beautiful. I think this is the kind of site we should have on this project!

**Comment by iks1** [comment]
@ZigRazor can we have meet regarding this, I want to clear some things... Maybe on discord ? Please let me know

**Comment by iks1** [comment]
@ZigRazor also if this design is final I would like to start the work and would request please create a new branch so that I can start working on it without effecting the current hosted branch also considering that current one is not working on my local machine.

**Comment by iks1** [comment]
> No, I don't have specific preferences about the tech stacks, the oly requirements is that works on github pages.

okay , I am starting then

**Comment by ZigRazor** [comment]
@iks1 Idon't know if you have already seen, but I have merged your PR

**Comment by iks1** [comment]
> @iks1 Idon't know if you have already seen, but I have merged your PR

Yeah I did , working on it, I was finalising some layout ... Will resume the work soon
